anvio/drivers/prodigal.py

```diff
@@ -21,11 +26,108 @@
 __maintainer__ = "A. Murat Eren"
 __email__ = "a.murat.eren@gmail.com"
 
-
 run = terminal.Run()
 progress = terminal.Progress()
 pp = terminal.pretty_print
 
+# How many times to retry the gene calling.
+NUM_RETRIES = 3
+
+
+def run_command_wrapper(command, log_file_path, run, retries=NUM_RETRIES):
+    """A wrapper around `utils.run_command`.
+
+    It will retry the command `retries` time.  Sometimes commands can fail for weird reasons so retrying often fixes
+    things.
+
+    If the return value of the command is zero, then it is returned.  If not, an instance of CommandError is returned.
+    """
+    for attempt in range(retries):
+        try:
+            ret_val = utils.run_command(command, log_file_path)
+        except ConfigError as e:
+            # utils.run_command will raise ConfigError on certain types of failures.
+            run.warning(f'Caught error: {e}, while running command {command}, attempt {attempt}, '
+                        'but we are going to keep trying!')
+            continue
+
+        # Return code of zero means success in the unix world.
+        if ret_val == 0:
+            return ret_val
+
+    # Failed to run the command a couple of times, so return an error.
+    return CommandError(f'Failed to run the command after {retries} tries. Final exit code was: {ret_val}')
+
+
+def split_fasta_wrapper(fasta_file_path, num_splits):
+    """A wrapper around `utils.split_fasta`.  It checks that the split files actually exist.
+
+    Returns output path.
+    """
+    paths = utils.split_fasta(fasta_file_path, parts=num_splits)
+
+    # Make sure all the splits actually exist.
+    for path in paths:
+        assert os.path.exists(path), f'I expected {path} to exist, but it did not.  Something weird happened.'
+
+    return paths
+
+
+def add_suffix(strings, suffix, separator='.'):
+    """Adds a suffix on to a list of `strings`.  Joins with `separator`."""
+    return [separator.join([string, suffix]) for string in strings]
+
+
+# Todo: Consider using shutil.copyfileobj instead
+def concatenate_files(intermediate_files, final_file):
```

Comment:
I think you should consider using `utils.concatenate_files` instead of this. the less the redundancy in the codebase the better for maintenance.

Reply:
Ah yep, that would be better...for some reason I did not check to see if that function already existed in the code base!